HealthMate/HealthMate.DAL/Repositories/GenericRepository.cs

```diff
@@ -13,20 +14,26 @@ public GenericRepository(ApplicationDbContext context)
             _context = context;
         }
 
-        public async Task<TEntity?> GetByIdAsync(Guid id, CancellationToken token) =>
-            await _context.Set<TEntity>().Where(t => t.Id.Equals(id)).SingleOrDefaultAsync(token);
+        public async Task<TEntity?> GetByIdAsync(Guid id,
+            CancellationToken token) =>
+            await _context.Set<TEntity>()
```

Comment:
I suggest you to use DbSet instead of _context.Set<TEntity>().
Just add new protected readonly field near the _context